src/lib/AutomateSDK.ts

```diff
@@ -54,50 +50,65 @@ export class AutomateSDK {
   private readonly _taskApi: Axios;
   private readonly _signature: Signature;
 
-  constructor(
+  private constructor(
+    chainId: number,
+    signer: Signer,
+    automate: Automate,
+    taskApi: Axios,
+    signature: Signature,
+  ) {
+    this._automateModule = new AutomateModule();
+    this._chainId = chainId;
+    this._signer = signer;
+    this._automate = automate;
+    this._taskApi = taskApi;
+    this._signature = signature;
+  }
+
+  public static async create(
     chainId: number,
     signer: Signer,
     signatureMessage?: string,
     config?: Partial<Config>,
-  ) {
+  ): Promise<AutomateSDK> {
     let automateAddress: string;
     if (config && config.isDevelopment) {
-      if (!isAutomateDevSupported(chainId)) {
+      if (!(await isAutomateDevSupported(chainId))) {
         throw new Error(`AutomateDev is not available on chainId:${chainId}`);
       }
 
       automateAddress = GELATO_ADDRESSES[chainId].automateDev!;
```

Comment:
The address should come from the API.

In think we should actually completely remove the constants `GELATO_ADDRESSES` as well as all static chain config, and instead make sure everything comes from the API